Prevent fatal errors on subscriptions table screen

james-allan opened the pull request:
## Description

When a subscription fails to load on the admin subscriptions table, instead of causing a fatal error by trying to call functions (like `get_status()`) on that incomplete subscription object, display just the ID and a tooltip which links to documentation with an explainer and troubleshooting steps.

## How to test this PR

It'll be hard to replicate this issue in a legitimate way given you need to have a corrupt subscription. The easiest way to do this might be to just go into the database and mess with one of your subscription's post data. eg setting your next payment date meta to a dummy string.

<img width="530.4" alt="Screen Shot 2022-01-05 at 11 56 36 am" src="https://user-images.githubusercontent.com/8490476/148148478-30691cb0-e408-403c-98b4-4768dbe83623.png">

1. After corrupting one or more of your subscriptions visit the admin **WooCommerce > Subscriptions** table.
    - On subscriptions-core trunk you will get a [fatal error](https://user-images.githubusercontent.com/8490476/148149271-774357ca-fb62-4092-b88d-f933275b73e9.png). 
    - On this branch of subscriptions-core you will see the subscription's [Post ID and a tooltip](https://user-images.githubusercontent.com/8490476/148148758-e6f4d795-ddda-4ef1-b093-8089f7a43f2e.png).
2. Clicking on the (?) tooltip will send you to this [accompanying documentation](https://woocommerce.com/document/subscriptions/store-manager-guide/#section-18) 

## Product impact
<!-- What products will this PR ship in? -->

- [x] Added changelog entry (or does not apply)
- [x] Will this PR affect WooCommerce Subscriptions?
    - 4286-gh-woocommerce/woocommerce-subscriptions
- [x] Will this PR affect WooCommerce Payments?
    - https://github.com/Automattic/woocommerce-payments/issues/3596

james-allan committed:
Only display the post ID if we cannot load a subscription on the admin subscriptions table

james-allan committed:
Only display the post ID if we cannot load a subscription on the admin subscriptions table

james-allan committed:
Fix typos and extra whitespace

james-allan committed:
Add changelog entry

aprea commented on includes/admin/class-wcs-admin-post-types.php:
```diff
@@ -445,6 +445,23 @@ public function render_shop_subscription_columns( $column ) {
+				echo '<strong>' . sprintf( esc_html_x( '#%s', 'hash before subscription number', 'woocommerce-subscriptions' ), esc_html( $post->ID ) ) . '</strong>';
+				?>
+				<div class="wcs-unknown-order-info-wrapper">
+					<a href="https://woocommerce.com/document/subscriptions/store-manager-guide/#section-18"><?php echo wcs_help_tip( sprintf( "This subscription couldn't be loaded from the database. %s Click to learn more.", '</br>' ) ); ?></a><?php // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>
```
**Nitpick:** Maybe add a `translators:` comment to explain that the `%s` = `<br>`.

Also, I think it should be `<br/>` or `<br>`, right now it looks like a closing tag, i.e. `</br>`

james-allan replied:
Oh good catch! I just realised this tip and [the one](https://woocommerce.com/document/subscriptions/orders/#section-8) this fix was inspired by, weren't being translated. I've fixed that in b20e492 and b4ee7fb.

aprea reviewed (approved):
This works for me ✅ Tested on both `trunk` and this branch and it works as expected. I left a comment above that might require a small change but other than that LGTM.

james-allan committed:
Add translator comment to unknown order and subscription comment

Wraps the content of these tooltips in i18n functions and replaces
the ordered parameter list with numbered placeholders

james-allan committed:
Reorganise the opening and closing PHP tags

james-allan commented:
Just making a note that after the changes above both the tips work as expected. 

<img width="494" alt="Screen Shot 2022-01-05 at 4 05 34 pm" src="https://user-images.githubusercontent.com/8490476/148168474-7fc2d9e4-ed4b-421f-9d9b-3a6f56b489ec.png">

<img width="226" alt="Screen Shot 2022-01-05 at 4 05 53 pm" src="https://user-images.githubusercontent.com/8490476/148168523-2a58ce73-7ddb-4d7f-a2d2-2560245d8cc8.png">
